Common/Source/Sound/ExtDev/Sound.cpp

```diff
@@ -76,21 +84,34 @@ void LKSound(const TCHAR *lpName) {
     const tstring& nmeaStr = _sound_table.getNmeaStr(sound_code);
 
     if (!nmeaStr.empty()) {
-        if (UseExtSound1) {
+        if (UseExtSound[0]) {
             devWriteNMEAString(devA(), nmeaStr.c_str());
         }
-        if (UseExtSound2) {
+        if (UseExtSound[1]) {
             devWriteNMEAString(devB(), nmeaStr.c_str());
         }
+        if (UseExtSound[2]) {
+            devWriteNMEAString(devC(), nmeaStr.c_str());
+        }
+        if (UseExtSound[3]) {
+            devWriteNMEAString(devD(), nmeaStr.c_str());
+        }
+        if (UseExtSound[4]) {
+            devWriteNMEAString(devE(), nmeaStr.c_str());
+        }
+        if (UseExtSound[5]) {
+            devWriteNMEAString(devF(), nmeaStr.c_str());
+        }
     }
         
 }
 
+
 void PlayResource (const TCHAR* lpName) {
 
     LKASSERT(lpName);
 
-    if(!lpName || !bSoundInit || !EnableSoundModes || (!UseExtSound1 && !UseExtSound2)) {
+    if(!lpName || !bSoundInit || !EnableSoundModes || (!ExtSound())) {
         return;
```

Comment:
why not move "if(!lpName || !bSoundInit || !EnableSoundModes)" inside "ExtSound" for avoid duplicated code ? (cf line 62)